src/client/connection.ts

```diff
@@ -71,12 +72,22 @@ export class Connection {
   }
 
   async sendMessageToServer(type: string, guid: string, method: string, params: any): Promise<any> {
+    const stackObject: any = {};
+    Error.captureStackTrace(stackObject);
+    const stack = stackObject.stack.startsWith('Error') ? stackObject.stack.substring(5) : stackObject.stack;
     const id = ++this._lastId;
     const validated = method === 'debugScopeState' ? params : validateParams(type, method, params);
-    const converted = { id, guid, method, params: validated };
+    const converted = { id, guid, method, params: validated, metadata: { stack } };
```

Comment:
Why can't we manage metadata on the client and not send it over rpc?

Reply:
This is a client-side stack that we want to see in the server-side tracing.